File: solutions/Leetcode_350/leetcode_350.cpp
#include <unordered_map>
#include <vector>

class Solution1
{
public:
    std::vector<int> intersect(std::vector<int> &nums1, std::vector<int> &nums2)
    {
        std::unordered_map<int, int> numDict;
        std::vector<int> intersection;
        for (int x : nums1)
            numDict[x]++;
        for (int x : nums2)
        {
            if (numDict.find(x) != numDict.end() && numDict[x] > 0)
            {
                intersection.push_back(x);
                numDict[x]--;
            }
        }
        return intersection;
    }
};

File: solutions/Leetcode_350/leetcode_350.py
from collections import defaultdict
from typing import List

class Solution1:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        intersection = []
        num_dict = defaultdict(int)
        for x in nums1:
            num_dict[x] += 1
        for x in nums2:
            if num_dict.get(x, 0):
                intersection.append(x)
                num_dict[x] -= 1
        return intersection
